### ISSUE TITLE
Severity filter is not taken into account

### ISSUE THREAD

**Issue by yguedidi** [friendsoftwig/twigcs#104]
When running `vendor/bin/twigcs -s "error" src/`, the warnings are still displayed

**Comment by OwlyCode** [comment]
I think there is a misunderstanding on how this works. The severity option defines the "maximum allowed error level" but does not control what is displayed, only the tool's exit code.

Here are some examples:

`twigcs -s notice ./dir`
- Shows all violations (notice, warning, errors)
- exit code `1` if any notice, warning or error. `0` otherwise

`twigcs -s warning ./dir`
- Shows all violations (notice, warning, errors)
- exit code `1` if any warning or error. `0` otherwise

`twigcs -s error ./dir`
- Shows all violations (notice, warning, errors)
- exit code `1` if any error. `0` otherwise

`twigcs -s ignore ./dir`
- Shows all violations (notice, warning, errors)
- exit code `0` whatever happens.

**But!** I gave a closer look at this option and there are two problems:

- The documentation says notices are not tolerated by default, but this actually is the case.
- The `ignore` value is not working as intended.

**Comment by yguedidi** [comment]
Way clearer to me now! Thank you!

Why I expected them to be hidden:
I was fixing the error violations, but I have too much warnings. It's a bit hard to fix the few errors I got in that big list, so I thought the severity option will help me filtering out the warning.
I tried `| grep ERROR` but I don't have the file name then.

What do you think about an new option `--show-allowed` (or `--hide-allowed`, depending on the default you prefer) that will filter the display?

**Comment by OwlyCode** [comment]
I see the use case. This would be a welcomed addition! Maybe `--severity-display-level` option would be more self explanatory?

**Comment by yguedidi** [comment]
My fear would be that it allows weird configuration, like `--severity warning --severity-display-level error`.
I can't imagine a case where the exit code will be `1` because of some warning violations while we don't display them.
Rephrased my use case is more "show me only the violations that raised the `1` exit code so that I can fix my code and make the CI green", not really "let me control what violations I see whatever the exit code is so that I fix violations level by level".
If the use case you'd like to solve is the second one, then indeed a dedicated level to what is displayed is better.

**Comment by OwlyCode** [comment]
Maybe the answer is to display by default only what causes a `1` exit code with `--severity` and then provide another option to show the "non blocking violations" (like `--display <all/blocking>`. That means we would only see what causes a blocking exit code for CI but we could still allow to display all so the CI is green but the non blocking violations are still tracked (by compiling stats based on reporting files).

**Comment by yguedidi** [comment]
Perfect! Completly agree with by default list only violations that causes the exit code 1.
On the way to alter the default behavior, me I'd prefer a flag `--show-allowed-violations`/`--display-non-blocking-violations`/something else, over an option that may be redundant with the default behavior.
But this is a matter of preference so I'm also OK with your `--display <all/blocking>` proposal

**Comment by stof** [comment]
@OwlyCode an solution could be to display the non-blocking violations only when running in verbose mode (showing only the total of non-blocking violations in non-verbose mode).

**Comment by bensquire** [comment]
@OwlyCode I was playing around with the codebase to see if I could put a PR together; and while this temporary hack works at only outputting errors; it's not what you're descirbing above.

```php
<?php

namespace Allocine\Twigcs\Reporter;

use Symfony\Component\Console\Output\OutputInterface;

class ConsoleReporter implements ReporterInterface
{
    /**
     * {@inheritdoc}
     */
    public function report(OutputInterface $output, array $violations)
    {
        $filename = null;
        $errorViolationCount = 0;
        
        /** @var \Allocine\Twigcs\Validator\Violation $violation */
        foreach ($violations as $violation) {
            if ($violation->getSeverity() < $violation::SEVERITY_ERROR) {
                continue;
            }

            $errorViolationCount++;

            if ($filename != $violation->getFilename()) {
                $filename = $violation->getFilename();
                $output->writeln(sprintf('<comment>%s</comment>', $filename));
            }

            $output->writeln(sprintf(
                '<comment>l.%d c.%d</comment> : %s %s',
                $violation->getLine(),
                $violation->getColumn(),
                $violation->getSeverityAsString(),
                $violation->getReason()
            ));
        }

        if ($errorViolationCount > 0) {
            $output->writeln(sprintf('<error>%d violation(s) found</error>', $errorViolationCount));
        } else {
            $output->writeln('<info>No violations found.</info>');
        }
    }
}
```

Any suggestions on how best to get the CLI options into the reporter so we can customise the output?

**Comment by OwlyCode** [comment]
@bensquire I think that if you want to implement what is described above you should do it inside the LintCommand class so you filter the violations before handing them to the reporter. This way each reporter is not forced to re-implement a filtering system and you can also have access to the command options (which bypasses the custom arguments in reporter class problem).

**Comment by OwlyCode** [comment]
Closing this as it is implemented by #125
